kernel/src/main/java/org/kframework/kprovex/ProofDefinitionBuilder.java

```diff
@@ -53,6 +60,7 @@ public Tuple2<Definition, Module> build(File specFile, String specModuleName, bo
         Map<String, Module> modulesMap = modules.stream().collect(Collectors.toMap(Module::name, m -> m));
         Definition parsedDefinition = compiledDefinition.getParsedDefinition();
         Module specModule = getModule(specModuleNameUpdated, modulesMap, parsedDefinition);
+        specModule = filter(specModule);
```

Comment:
Do we need to apply it to _every_ module that is not in the `--main-module`? Will this only apply it to the `--spec-module`?

If it's not the case (will apply to all modules), feel free to automerge. The tests make me think that it should work on all the modules with claims in them.

Reply:
This step applies recursively to all modules imported by the spec module so nothing is left behind.